kubernetes/client.go

```diff
@@ -211,3 +211,26 @@ func filterPodsForService(selector labels.Selector, allPods *v1.PodList) []v1.Po
 	}
 	return pods
 }
+
+// filterPodsForEndpoints performs a second pass was selector may return too many data
+// This case happens when a "nil" selector (such as one of default/kubernetes service) is used
+func filterPodsForEndpoints(endpoints *v1.Endpoints, unfiltered *v1.PodList) []v1.Pod {
```

Comment:
But, if we know the endpoints, can't we pass the list of endpoints as a selector/filter in the GetPods logic ?
That would rely in the k8s backend to filter the expected pods, isn't it ?